workflow/artifacts/git/git.go

```diff
@@ -27,7 +29,20 @@ type GitArtifactDriver struct {
 	InsecureIgnoreHostKey bool
 }
 
-func (g *GitArtifactDriver) auth() (func(), transport.AuthMethod, []string, error) {
+var (
+	sshURLRegex = regexp.MustCompile("^(ssh://)?([^/:]*?)@[^@]+$")
+)
+
+// IsSSHURL returns true if supplied URL is SSH URL - referenced https://github.com/argoproj/argo-cd/blob/d516f4754f4ddf99ee383452d4ff529f1cc0530c/util/git/git.go#L73
+func IsSSHURL(url string) (bool, string) {
```

Comment:
Two things:
* Don't export this function (begin function name with lower case)
* Return `(string, bool)` fits convention

```suggestion
func isSSHURL(url string) (string, bool) {
```